plexapi/audio.py

```diff
@@ -125,6 +132,35 @@ def sync(self, bitrate, client=None, clientId=None, limit=None, title=None):
 
         return myplex.sync(sync_item, client=client, clientId=clientId)
 
+    def sonicallySimilar(
+        self: TAudio,
+        limit: Optional[int] = None,
+        maxDistance: Optional[float] = None,
+        **kwargs,
+    ) -> List[TAudio]:
+        """Returns a list of sonically similar audio items.
+
+        Parameters:
+            limit (int): Maximum count of items to return. Default 50 (server default)
+            maxDistance (float): Maximum distance between tracks, 0.0 - 1.0. Default 0.25 (server default).
+            **kwargs: Additional options passed into :func:`~plexapi.base.PlexObject.fetchItems`.
+
+        Returns:
+            List[:class:`~plexapi.audio.Audio`]: list of sonically similar audio items.
+        """
+
+        key = f"{self.key}/nearest"
+        params = {"maxDistance": maxDistance, "limit": limit}
+        key += utils.joinArgs(
+            {k: v for k, v in params.items() if v is not None}
+        )
```

Comment:
It is more clear to add params instead of removing params.
```suggestion
        params = {}
        if limit is not None:
            params['limit'] = limit
        if maxDistance is not None:
            params['maxDistance'] = maxDistance
        key += utils.joinArgs(params)
```

Reply:
I have a way to maintain the adding of param as well as make it add programmatically so that it is scalable and avoid code repetition of `if` statements for every key

```py
        all_args = {"limit": limit, "maxDistance": maxDistance}
        params = {k: v for k, v in all_args.items() if v is not None}  # add if not None


```